core_lib/src/managers/selectionmanager.cpp

```diff
@@ -0,0 +1,370 @@
+#include "selectionmanager.h"
+#include "editor.h"
+
+#include "layerbitmap.h"
+#include "vectorimage.h"
+#include "bitmapimage.h"
+
+#include "layervector.h"
+
+//#ifdef QT_DEBUG
+#include <QDebug>
+//#endif
+
+
+SelectionManager::SelectionManager(Editor* editor) : BaseManager(editor),
+    mySelection(QRectF()), myTempTransformedSelection(QRectF()), myTransformedSelection(QRectF()),
+    mSomethingSelected(false), mLastSelectionPolygonF(QPolygonF()), mCurrentSelectionPolygonF(QPolygonF()),
+    myRotatedAngle(0), mOffset(QPointF()), mMoveMode(MoveMode::NONE)
```

Comment:
Not really, because it's all happening prior to the constructor being called. It's the alternative to initialising in the header, member functions can't be called yet.

Reply:
Well I meant in the constructor, but it doesn't really matter. I mean all most of these values are the default values, so I guess it's better to leave them here for documentation purposes.